fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/QuantifiedObjectValue.java

```diff
@@ -89,7 +89,11 @@ public Value replaceReferenceWithField(@Nonnull final FieldValue fieldValue) {
     @Override
     public <M extends Message> Object eval(@Nullable final FDBRecordStoreBase<M> store, @Nonnull final EvaluationContext context) {
         // TODO this "if" can be encoded in encapsulation code implementing type promotion rules
-        final var binding = (QueryResult)context.getBinding(Bindings.Internal.CORRELATION, alias);
+        final var obj = context.getBinding(Bindings.Internal.CORRELATION, alias);
```

Comment:
This his more a note for myself: I wonder if this clashes in concept with `QuantifierRecordValue`s as being introduced in the implicit group by PR.